core-processor/src/ext.rs

```diff
@@ -768,6 +768,19 @@ impl CountersOwner for Ext {
     }
 }
 
+impl Ext {
+    fn atomic_gas<T, E>(&mut self, func: impl FnOnce() -> Result<T, E>) -> Result<T, E> {
+        let amount = self.context.gas_counter;
+
+        func().inspect_err(|_| {
```

Comment:
We tend to use stable compiler, therefore maybe it would be better to use `map_err` instead?

Reply:
Resolved in 0212aa1.